nmdc_automation/re_iding/scripts/re_id_tool.py

```diff
@@ -691,6 +618,402 @@ def process_records(ctx, study_id, data_dir, update_links=False, identifiers_fil
+            else:
+                logging.info(f"Workflow ID in the database is correct: {workflow_id}")
+
+            # Data Objects and corresponding Data File Paths
+            for data_object in record["data_objects"]:
+                data_object_id = data_object["id"]
+                data_object_url = data_object["url"]
+                data_object_name = data_object["name"]
+                data_object_type = data_object.get("data_object_type")
+                # check the name and url and add to the updates_map if they are malformed
+                fixed_data_object_name = fix_malformed_data_object_name(data_object_name)
+                fixed_data_object_url = fix_malformed_data_object_url(data_object_url)
+                if fixed_data_object_name != data_object_name:
+                    logging.warning(f"Data Object name is malformed: {data_object_name}")
+                    updates_map.setdefault("data_object_set", []).append({
+                        "q": {"id": data_object_id},
+                        "u": {"$set": {"name": fixed_data_object_name}}
+                    })
+                if fixed_data_object_url != data_object_url:
+                    logging.warning(f"Data Object URL is malformed: {data_object_url}")
+                    logging.info(f"Fixed URL: {fixed_data_object_url}")
+                    updates_map.setdefault("data_object_set", []).append({
+                        "q": {"id": data_object_id},
+                        "u": {"$set": {"url": fixed_data_object_url}}
+                    })
+
+            # Workflow Directories
+            # Special case:
+            #   - No affected data objects and multiple workflow directories with incorrect blades - skip for now
+            if len(record["workflow_dirs"]) > 1 and not record["data_objects"]:
+                logging.warning(f"Multiple workflow directories found for workflow ID: {workflow_id}")
+                # if none of the workflow directories contain the correct blade, we have a problem that we can't fix here
+                if not any([wf_blade in workflow_dir for workflow_dir in record["workflow_dirs"]]):
+                    logging.error(f"None of the workflow directories contain the correct blade: {wf_blade}")
+                    continue
+            # Workflow directories: check for malformed directory names and fix them. Add the old ID to alternative_identifiers
+            # and create a symbolic link to the new directory
+            for workflow_dir in record["workflow_dirs"]:
+                workflow_dir_name = workflow_dir.split("/")[-1]
+                omics_dir_name = workflow_dir.split("/")[-2]
+                fixed_workflow_dir_name = fix_malformed_workflow_id_version(workflow_dir_name)
+                # get the actual data file path depending on the environment
+                workflow_dir_path = data_dir.joinpath(omics_dir_name, workflow_dir_name)
+                if fixed_workflow_dir_name != workflow_dir_name:
```

Comment:
If this block is for mismatched blades and the ID was only ever used on the filesystem we don't need to store that as an alternative ID